media_driver/linux/common/vp/ddi/media_libva_vp.c

```diff
@@ -4208,6 +4213,12 @@ DdiVp_QueryVideoProcFilterCaps (
                 if (uQueryFlag == QUERY_CAPS_ATTRIBUTE)
                 {
                     VAProcFilterCapHighDynamicRange *HdrTmCap = (VAProcFilterCapHighDynamicRange *)filter_caps;
+
+                    if (uQueryCapsNum < uExistCapsNum)
+                    {
+                        return VA_STATUS_ERROR_MAX_NUM_EXCEEDED;
+                    }
+
```

Comment:
I believe there is one more thing which needs to be changed in this code. Basically, `num_filter_caps` gets assigned even in those cases when function will return an error. This is bad style especially when this can be avoided like here. So, I suggest to avoid assignment of num_filter_caps before we processed all error checks. And we actually can move assign out of the switch:
```
uint32_t uExistCapsNum = 0;  // please, add initialization
switch (type)
{
... // assign uExistCapsNum here
}

*num_filter_caps = uExistCapsNum;
```

Reply:
`num_filter_caps` is assigned before error checking in the original code, but I agree with you that we'd better to move the assign out of the switch.

Reply:
According to https://github.com/intel/libva/blob/d49b99d5d88f66e2e1ca451cf19251c7950c870d/va/va_vpp.h#L1408-L1426, num_filter_caps must be adjusted when VA_STATUS_ERROR_MAX_NUM_EXCEEDED is returned, so `num_filter_caps` should get assigned before error checking.